src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/Regex.CompileToAssembly.Tests.cs

```diff
@@ -11,27 +13,66 @@ namespace System.Text.RegularExpressions.Tests
                 new AssemblyName("abcd")));
 
             Assert.Throws<PlatformNotSupportedException>(() => Regex.CompileToAssembly(
-                new[] { new RegexCompilationInfo("abcd", RegexOptions.None, "abcd", "SomeNamespace", true) },
+                [new RegexCompilationInfo("abcd", RegexOptions.None, "abcd", "SomeNamespace", true)],
                 new AssemblyName("abcd"),
-                new[] { new CustomAttributeBuilder(typeof(AssemblyCompanyAttribute).GetConstructor(new[] { typeof(string) }), new[] { "TestCompany" }) }));
+                [new CustomAttributeBuilder(typeof(AssemblyCompanyAttribute).GetConstructor([typeof(string)]), new[] { "TestCompany" })]));
 
             Assert.Throws<PlatformNotSupportedException>(() => Regex.CompileToAssembly(
-                new[] { new RegexCompilationInfo("abcd", RegexOptions.None, "abcd", "SomeNamespace", true) },
+                [new RegexCompilationInfo("abcd", RegexOptions.None, "abcd", "SomeNamespace", true)],
                 new AssemblyName("abcd"),
-                new[] { new CustomAttributeBuilder(typeof(AssemblyCompanyAttribute).GetConstructor(new[] { typeof(string) }), new[] { "TestCompany" }) },
+                [new CustomAttributeBuilder(typeof(AssemblyCompanyAttribute).GetConstructor([typeof(string)]), new[] { "TestCompany" })],
                 "resourceFile"));
         }
+
+
+        [ConditionalFact(nameof(IsDebug))]
+        public void CompileToAssembly_SimpleUseInDebug()
+        {
+            (RegexCompilationInfo rci, string validInput, string invalidInput)[] regexes =
+            [
+                (new RegexCompilationInfo("abcd", RegexOptions.None, "Type1", "Namespace1", ispublic: true), "123abcd123", "123abed123"),
+                (new RegexCompilationInfo("(a|b|cde)+", RegexOptions.None, "Type2", "Namespace2.Sub", ispublic: true), "abcde", "cd"),
+            ];
+
+            string assemblyName = Path.GetRandomFileName();
+
+            string cwd = Environment.CurrentDirectory;
+            Environment.CurrentDirectory = TestDirectory;
+            try
+            {
+                Regex.CompileToAssembly(regexes.Select(r => r.rci).ToArray(), new AssemblyName(assemblyName));
+            }
+            finally
+            {
+                Environment.CurrentDirectory = cwd;
+            }
+
+            string assemblyPath = Path.Combine(TestDirectory, assemblyName + ".dll");
+
+            // TODO https://github.com/dotnet/runtime/issues/15704: Currently fails with Format of the executable (.exe) or library (.dll) is invalid.
+
+            //Assembly a = Assembly.LoadFrom(assemblyPath);
+            //foreach ((RegexCompilationInfo rci, string validInput, string invalidInput) in regexes)
+            //{
+            //    Regex r = (Regex)Activator.CreateInstance(a.GetType($"{rci.Namespace}.{rci.Name}"));
+            //    Assert.True(r.IsMatch(validInput));
+            //    Assert.False(r.IsMatch(invalidInput));
+            //}
```

Comment:
@buyaa-n, this fails with a load exception stating "Currently fails with Format of the executable (.exe) or library (.dll) is invalid".  Is that expected at this point?  You can uncomment this and run the regex function tests to diagnose if it's not expected.

Reply:
This should have fixed with this PR https://github.com/dotnet/runtime/pull/96362 that merged yesterday, not sure if its using the fixed version ...

Reply:
When I run with the latest System.Reflection.Emit, then I get this:
```C#
Process terminated. Assertion failed.
     at System.Reflection.Emit.ILGeneratorImpl.AdjustDepth(OpCode opcode, LabelInfo label) in D:\repos\runtime\src\libraries\System.Reflection.Emit\src\System\Reflection\Emit\ILGeneratorImpl.cs:line 444
     at System.Reflection.Emit.ILGeneratorImpl.Emit(OpCode opcode, Label label) in D:\repos\runtime\src\libraries\System.Reflection.Emit\src\System\Reflection\Emit\ILGeneratorImpl.cs:line 475
     at System.Text.RegularExpressions.RegexCompiler.BrtrueFar(Label l) in D:\repos\runtime\src\libraries\System.Text.RegularExpressions\src\System\Text\RegularExpressions\RegexCompiler.cs:line 268
     at System.Text.RegularExpressions.RegexCompiler.EmitScan(RegexOptions options, MethodInfo tryFindNextStartingPositionMethod, MethodInfo tryMatchAtCurrentPositionMethod) in D:\repos\runtime\src\libraries\System.Text.Regular
  Expressions\src\System\Text\RegularExpressions\RegexCompiler.cs:line 5403
     at System.Text.RegularExpressions.RegexAssemblyCompiler.GenerateRegexType(String pattern, RegexOptions options, String name, Boolean isPublic, RegexTree tree, RegexInterpreterCode code, TimeSpan matchTimeout) in D:\repos\r
  untime\src\libraries\System.Text.RegularExpressions\src\System\Text\RegularExpressions\RegexAssemblyCompiler.cs:line 66
     at System.Text.RegularExpressions.Regex.CompileToAssembly(RegexCompilationInfo[] regexinfos, AssemblyName assemblyname, CustomAttributeBuilder[] attributes, String resourceFile) in D:\repos\runtime\src\libraries\System.Tex
  t.RegularExpressions\src\System\Text\RegularExpressions\Regex.cs:line 256
     at System.Text.RegularExpressions.Regex.CompileToAssembly(RegexCompilationInfo[] regexinfos, AssemblyName assemblyname) in D:\repos\runtime\src\libraries\System.Text.RegularExpressions\src\System\Text\RegularExpressions\Re
  gex.cs:line 223
     at System.Text.RegularExpressions.Tests.RegexCompileToAssemblyTests.CompileToAssembly_SimpleUseInDebug() in D:\repos\runtime\src\libraries\System.Text.RegularExpressions\tests\FunctionalTests\Regex.CompileToAssembly.Tests.
  cs:line 58
```
It's tripping over this assert:
https://github.com/dotnet/runtime/blob/5a2d1519b01b2935e343b6aba2989f97ed596914/src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs#L444

Reply:
And if I disable the assert, then when I try to use the assembly, I get:
```
  Unhandled exception. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
   ---> System.InvalidProgramException: Common Language Runtime detected an invalid program.
     at Namespace1.Type1..ctor()
     at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions) in D:\repos\runtime\src\coreclr\System.Private.CoreLib\src\System\RuntimeType.CoreCLR.cs:line 3914
```

Reply:
Hm, I wonder if the code trying to run the assembly created (invoke methods, create instance), the new `AssemblyBuilder` is only for saving, after saving you can load the saved assembly with `Assembly.LoadFrom(assemblyPath);` or so and run.

Reply:
Actually, the latter looks like user error on my part, and the presence of the invalid IL was triggering the assert. We should ensure we don't have asserts that fire with invalid IL.